url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/appliances/ConsumptionModel.java

```diff
@@ -0,0 +1,358 @@
+				totalDuration += (n[i] * t.d);
+			}
+		}
+	}
+
+	/**
+	 * Gets the total duration of the consumption model.
+	 *
+	 * @return the total duration of the consumption model
+	 */
+	public int getTotalDuration() { return totalDuration; }
+
+	/**
+	 * Gets how many times the patterns repeats.
+	 *
+	 * @return how many times the patterns repeats
+	 */
+	public int getOuterN() { return outerN; }
+
+	/**
+	 * Gets the number of patterns.
+	 *
+	 * @return the number of patterns
+	 */
+	public int getPatternN() { return patternN; }
+
+	/**
+	 * Gets the number of times that the i-th pattern runs.
+	 *
+	 * @param i the index i
+	 * @return the the number of times that the i-th pattern runs
+	 */
+	public int getN(int i) { return n[i]; }
+
+	/**
+	 * Gets the i-th pattern's duration.
+	 *
+	 * @param i the index i
+	 * @return the i-th pattern's duration
+	 */
+	public int getPatternDuration(int i) { return patternDuration[i]; }
+
+	/**
+	 * Gets the i-th pattern.
+	 *
+	 * @param i the index i
+	 * @return the i-th pattern
+	 */
+	public ArrayList<Tripplet> getPattern(int i) { return patterns[i]; }
+
+	/**
+	 * Gets the array storing the consumption patterns.
+	 *
+	 * @return the array storing the consumption patterns
+	 */
+	public ArrayList<Tripplet>[] getPatterns() { return patterns; }
+
+	/**
+	 * Check if consumption is static.
+	 *
+	 * @return true, if consumption is static.
+	 */
+	public boolean checkStatic()
+	{
+		boolean result = true;
+
+		Double[] values = getValues();
+
+		// System.out.println("Appliance: " + name + " Model: "
+		// + activeConsumptionModelString);
+		//
+		// System.out.println("Appliance: " + name + " Values: "
+		// + Arrays.toString(values));
+
+		for (int i = 0; i < values.length - 1; i++) {
+			// System.out.println("Previous: " + values[i].doubleValue() + " Next: "
+			// + values[i + 1].doubleValue());
+			if (values[i].doubleValue() != values[i + 1].doubleValue()) {
+				// System.out.println("IN");
+				result = false;
+				break;
+			}
+		}
+
+		return result;
+	}
+
+	/**
+	 * Gets the consumption.
+	 *
+	 * @return the consumption
+	 */
+	public Double[] getConsumption()
+	{
+
+		ArrayList<Double> temp = new ArrayList<Double>();
+		int times = getOuterN();
+		if (times == 0)
+			times = 2;
+		// Number of repeats
+		for (int i = 0; i < times; i++) {
+			// System.out.println("Time: " + i);
+			// Number of patterns in each repeat
+			for (int j = 0; j < getPatternN(); j++) {
+				// System.out.println("Pattern: " + j);
+				int internalTimes = getN(j);
+				if (internalTimes == 0)
+					internalTimes = 2;
+				// System.out.println("Internal Times: " + k);
+				for (int k = 0; k < internalTimes; k++) {
+					ArrayList<Tripplet> tripplets = getPattern(j);
+					for (int l = 0; l < tripplets.size(); l++) {
+						// System.out.println("TripletPower: " + l);
+						for (int m = 0; m < tripplets.get(l).d; m++) {
+							temp.add(tripplets.get(l).v);
+						}
+					}
+				}
+			}
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 193:212, 326:345.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.